comp/syntheticstestscheduler/impl/scheduler.go

```diff
@@ -0,0 +1,236 @@
+		runTraceroute:                runTraceroute,
+	}
+
+	// by default, sendResult delegates to the real forwarder-backed implementation
+	scheduler.sendResult = scheduler.sendSyntheticsTestResult
+
+	scheduler.ticker = time.NewTicker(scheduler.flushInterval)
+	scheduler.tickerC = scheduler.ticker.C
+
+	if err := scheduler.retrieveConfig(); err != nil {
+		return nil, err
+	}
+	return scheduler, nil
+}
+
+// runningTestState represents in-memory runtime data for a scheduled test.
+type runningTestState struct {
+	cfg     common.SyntheticsTestConfig
+	lastRun time.Time
+	nextRun time.Time
+}
+
+type runningState struct {
+	mu    sync.RWMutex
+	tests map[string]*runningTestState // PublicID -> runtime state
+}
+
+// onConfigUpdate handles remote-config updates for synthetics tests.
+func (s *SyntheticsTestScheduler) onConfigUpdate(updates map[string]state.RawConfig, applyStateCallback func(string, state.ApplyStatus)) {
+	s.log.Debugf("Updates received: count=%d", len(updates))
+
+	newConfig := map[string]common.SyntheticsTestConfig{}
+	for configPath, rawConfig := range updates {
+		s.log.Debugf("received config %s: %s", configPath, string(rawConfig.Config))
+		syntheticsTestCfg := common.SyntheticsTestConfig{}
+		if err := json.Unmarshal(rawConfig.Config, &syntheticsTestCfg); err != nil {
+			s.log.Warnf("skipping invalid Synthetics Test update %s: %v", configPath, err)
+			applyStateCallback(configPath, state.ApplyStatus{
+				State: state.ApplyStateError,
+				Error: "error unmarshalling payload",
+			})
+			continue
+		}
+
+		newConfig[syntheticsTestCfg.PublicID] = syntheticsTestCfg
+		s.log.Debugf("Processed config %s: %v", configPath, syntheticsTestCfg)
+
+		applyStateCallback(configPath, state.ApplyStatus{State: state.ApplyStateAcknowledged})
+	}
+
+	s.updateRunningState(newConfig)
+	s.runningConfig = newConfig
+
+	if err := s.persistConfig(); err != nil {
+		s.log.Warnf("unable to persist synthetics test config: %v", err)
+		// TODO: what path should I provide if it is global to the config
+		applyStateCallback("TODO", state.ApplyStatus{
+			State: state.ApplyStateError,
+			Error: "unable to persist synthetics test config",
+		})
+	}
+}
+
+// updateRunningState synchronizes in-memory runtime state with a new configuration.
+func (s *SyntheticsTestScheduler) updateRunningState(newConfig map[string]common.SyntheticsTestConfig) {
+	s.state.mu.Lock()
+	defer s.state.mu.Unlock()
+
+	seen := map[string]struct{}{}
+	for _, newTestConfig := range newConfig {
+		pubID := newTestConfig.PublicID
+		seen[pubID] = struct{}{}
+		current, exists := s.state.tests[pubID]
+		if !exists {
+			s.state.tests[pubID] = &runningTestState{
+				cfg:     newTestConfig,
+				lastRun: time.Time{},
+				nextRun: s.TimeNowFn().Add(time.Duration(newTestConfig.Interval) * time.Second),
+			}
+		} else {
+			current.cfg = newTestConfig
+		}
+	}
+
+	for pubID := range s.state.tests {
+		if _, exists := seen[pubID]; !exists {
+			delete(s.state.tests, pubID)
+		}
+	}
```

Comment:
Are we sure this logic works? 

I mean, if users have many RC configs, they might *not arrive all in one RC update* right?
(actually, I'm not fully sure, but something to confirm with RC team if not already).

If so, this logic won't work, since the `seen` map here is specific to this case only.